resolver/addressable_resolver.go

```diff
@@ -211,3 +211,109 @@ func (r *URIResolver) URIFromObjectReference(ctx context.Context, ref *corev1.Ob
 	}
 	return url, nil
 }
+
+// AddressableFromDestination resolves a v1beta1.Destination into a addressable.
+func (r *URIResolver) AddressableFromDestination(ctx context.Context, dest duckv1beta1.Destination, parent interface{}) (*duckv1.Addressable, error) {
+	var deprecatedObjectReference *corev1.ObjectReference
+	if !(dest.DeprecatedAPIVersion == "" && dest.DeprecatedKind == "" && dest.DeprecatedName == "" && dest.DeprecatedNamespace == "") {
+		deprecatedObjectReference = &corev1.ObjectReference{
+			Kind:       dest.DeprecatedKind,
+			APIVersion: dest.DeprecatedAPIVersion,
+			Name:       dest.DeprecatedName,
+			Namespace:  dest.DeprecatedNamespace,
+		}
+	}
+	var addr *duckv1.Addressable
+	if dest.Ref != nil && deprecatedObjectReference != nil {
+		return addr, errors.New("ref and [apiVersion, kind, name] can't be both present")
+	}
+	var ref *corev1.ObjectReference
+	if dest.Ref != nil {
+		ref = dest.Ref
+	} else {
+		ref = deprecatedObjectReference
+	}
+
+	if ref != nil {
+		addr, err := r.AddressableFromObjectReference(ctx, dest, parent)
+		if err != nil {
+			return addr, err
+		}
+		return addr, nil
+	}
+
+	return addr, errors.New("destination missing Ref, [apiVersion, kind, name] and URI, expected at least one")
+}
+
+// AddressableFromDestination resolves a v1.Destination into a addressable.
+func (r *URIResolver) AddressableFromDestinationV1(ctx context.Context, dest duckv1.Destination, parent interface{}) (*duckv1.Addressable, error) {
+	var addr *duckv1.Addressable
+	if dest.Ref != nil {
+		addr, err := r.AddressableFromKReference(ctx, dest, parent)
+		if err != nil {
+			return nil, err
+		}
+		return addr, nil
+	}
+
+	return addr, errors.New("destination missing Ref, [apiVersion, kind, name] and URI, expected at least one")
+}
+
+func (r *URIResolver) AddressableFromKReference(ctx context.Context, dest duckv1.Destination, parent interface{}) (*duckv1.Addressable, error) {
+	kRef := dest.Ref
+	ref := &corev1.ObjectReference{Name: kRef.Name, Namespace: kRef.Namespace, APIVersion: kRef.APIVersion, Kind: kRef.Kind}
+
+	gvr, _ := meta.UnsafeGuessKindToResource(ref.GroupVersionKind())
+	if err := r.tracker.TrackReference(tracker.Reference{
+		APIVersion: ref.APIVersion,
+		Kind:       ref.Kind,
+		Namespace:  ref.Namespace,
+		Name:       ref.Name,
+	}, parent); err != nil {
+		return nil, fmt.Errorf("failed to track reference %s %s/%s: %w", gvr.String(), ref.Namespace, ref.Name, err)
+	}
```

Comment:
@pierDipi Is this necessary, because `duckv1.Destination`'s `KReference` doesn't have `ref.GroupVersionKind()`  so I have to create a `corev1.ObjectReference` from `KReference` and this doesn't look right.